Source/WebCore/style/InlineTextBoxStyle.cpp

```diff
@@ -181,6 +181,17 @@ static float computedUnderlineOffset(const UnderlineOffsetArguments& context)
         computedUnderlineOffset = std::max<float>(desiredOffset, fontMetrics.ascent());
         break;
     }
+    case TextUnderlinePosition::Left:
+    case TextUnderlinePosition::Right: {
+        if (context.lineStyle.isHorizontalWritingMode()) {
+            ASSERT(context.textUnderlinePositionUnder);
+            // Position underline relative to the bottom edge of the lowest element's content box.
+            auto desiredOffset = context.textUnderlinePositionUnder->textRunLogicalHeight + gap + std::max(context.textUnderlinePositionUnder->textRunOffsetFromBottomMost, 0.f) + underlineOffset.lengthOr(0.f);
+            computedUnderlineOffset = std::max<float>(desiredOffset, fontMetrics.ascent());
+        } else
```

Comment:
This looks like the case above this one (TextUnderlinePosition::Under). It is possibly to do some code sharing here?